client/protocol.go

```diff
@@ -129,10 +129,29 @@ func fastFetchCluster(addrs []string, proposeTimeout time.Duration) (*curpapi.Fe
 
 // Propose the request to servers, if use_fast_path is false, it will wait for the synced index
 func (c *protocolClient) Propose(cmd *xlineapi.Command, useFastPath bool) (*proposeRes, error) {
-	if useFastPath {
-		return c.fastPath(cmd)
+	var res *proposeRes
+	var err error
+	for {
+		if useFastPath {
+			res, err = c.fastPath(cmd)
+		} else {
+			res, err = c.slowPath(cmd)
+		}
+
+		if err != nil {
+			cluster, err := c.fetchCluster(false)
+			if err != nil {
+				return nil, err
+			}
+			err = c.setCluster(cluster)
+			if err != nil {
+				return nil, err
+			}
+			time.Sleep(2 * time.Second)
+			continue
+		}
```

Comment:
Please check err type, not every err needs a fetch cluster.

Reply:
Done.